python/tk_multi_publish2/dialog.py

```diff
@@ -40,7 +40,98 @@ class AppDialog(QtGui.QWidget):
     (DRAG_SCREEN, PUBLISH_SCREEN) = range(2)
 
     # details ui panes
-    (ITEM_DETAILS, TASK_DETAILS, PLEASE_SELECT_DETAILS) = range(3)
+    (ITEM_DETAILS, TASK_DETAILS, PLEASE_SELECT_DETAILS, MULTI_EDIT_NOT_SUPPORTED) = range(4)
+
+    # settings ui panes, those are both present on the TASK_DETAILS pane.
+    (BUILTIN_TASK_DETAILS, CUSTOM_TASK_DETAILS) = range(2)
+
+    class TaskSelection(object):
+        """
+        Allows to manipulate a task selection as if it was a single object.
+
+        This class assumes that every task is of the same plugin type.
+
+        :param items: List of task for in the selection. Defaults to an empty list.
+        """
+        def __init__(self, items=None):
+            self._items = items or []
+
+        def is_same_task_type(self, task_selection):
+            """
+            Checks if this task selection is of the same type as another task selection.
+
+            :param task_selection: List of :class:`Task`.
+
+            :returns: ``True`` is the plugins are the same, ``False`` otherwise. Note that
+                two empty lists are considered to be of the same type.
+            """
+            if self._items and task_selection._items:
+                return self._items[0].is_same_task_type(task_selection._items[0])
```

Comment:
Maybe a quick comment here that explains why we only need to check the first item in each selection.

Reply:
Also, maybe this selection object should just expose the type of `_item[0]`, similar to the `plugin` property.

Then you could write:

```python
return self.type == task_selection.type
```